.github/workflows/publish-docker.yml

```
@@ -0,0 +1,133 @@
+name: Create and publish a Docker image
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# Configures this workflow to run every time a change is pushed to the branch called `release`.
+on:
+  push:
+    branches: ["main"]
+    tags: ["v*"]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  DOCKERFILE: ./Dockerfile
+  REGISTRY_LOGIN: ${{ github.actor }}
+  REGISTRY_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+  docker_build:
+    strategy:
+      fail-fast: true
+      matrix:
+        os: ["ubuntu-22.04", "ubuntu-22.04-arm"]
```

```suggestion
        os: ["ubuntu-24.04", "ubuntu-24.04-arm"]
```
we're using 24.04 everywhere in the build, I think